lib/AnimationStation/src/Effects/CustomTheme.cpp

```diff
@@ -1,41 +1,108 @@
 #include "CustomTheme.hpp"
 
 std::map<uint32_t, RGB> CustomTheme::theme;
+std::map<uint32_t, int32_t> CustomTheme::times = {};
+std::map<uint32_t, RGB> CustomTheme::hitColor = {};
 
 CustomTheme::CustomTheme(PixelMatrix &matrix) : Animation(matrix) {
+    for (size_t r = 0; r != matrix.pixels.size(); r++) {
+        for (size_t c = 0; c != matrix.pixels[r].size(); c++) {
+            if (matrix.pixels[r][c].index == NO_PIXEL.index)
+                continue;
+            times.insert_or_assign(matrix.pixels[r][c].index, 0);
+            hitColor.insert_or_assign(matrix.pixels[r][c].index, defaultColor);            
+        }
+    }
+}
 
+void CustomTheme::UpdatePixels(std::vector<Pixel> inpixels) {
+  this->pixels = inpixels;
 }
 
 void CustomTheme::Animate(RGB (&frame)[100]) {
-  for (size_t r = 0; r != matrix->pixels.size(); r++) {
-    for (size_t c = 0; c != matrix->pixels[r].size(); c++) {
-      if (matrix->pixels[r][c].index == NO_PIXEL.index)
-        continue;
-
-      auto itr = theme.find(matrix->pixels[r][c].mask);
-      if (itr != theme.end())
-        for (size_t p = 0; p != matrix->pixels[r][c].positions.size(); p++)
-          frame[matrix->pixels[r][c].positions[p]] = itr->second;
-      else
-        for (size_t p = 0; p != matrix->pixels[r][c].positions.size(); p++)
-          frame[matrix->pixels[r][c].positions[p]] = defaultColor;
+
+    coolDownTimeInMs = AnimationStation::options.customThemeCooldownTimeInMs;
```

Comment:
and we can just do assignments before logic at the top. its a little easier to read. so like


```
uint32_t now = get_absolute_time();
coolDownTimeInMs = AnimationStation::options.customThemeCooldownTimeInMs;
updateTimeInms = absolute_time_diff_us(lastUpdateTime, now) / 1000;

... 

lastUpdateTime = now;
```

that makes it so you don't have to call the Pico SDK API for time 3 times in this function and assignment for those two variables is at the top